ibm/resource_ibm_cis_dns_record.go

```diff
@@ -444,52 +569,453 @@ func resourceIBMCISDnsRecordUpdate(d *schema.ResourceData, meta interface{}) err
 				newDataMap[id] = newData
 			}
 
-			updateRecord.Data = newDataMap
+			opt.SetData(newDataMap)
 		}
+
 		if contentOk == dataOk {
 			return fmt.Errorf(
 				"either 'content' (present: %t) or 'data' (present: %t) must be provided",
 				contentOk, dataOk)
 		}
 
-		if err := validateRecordName(updateRecord.DnsType, updateRecord.Content); err != nil {
-			return fmt.Errorf("Error validating record name %q: %s", updateRecord.Name, err)
+		if priority, ok := d.GetOk("priority"); ok {
+			opt.SetPriority(priority.(int64))
+		}
+		if ttl, ok := d.GetOk("ttl"); ok {
+			opt.SetTTL(ttl.(int64))
 		}
 
-		_, err = cisClient.Dns().UpdateDns(cisID, zoneID, dnsID, updateRecord)
+	}
+
+	result, response, err := sess.CreateDnsRecord(opt)
+	if err != nil {
+		log.Printf("Error creating dns record: %s, error %s", response, err)
+		return err
+	}
+
+	d.SetId(convertCisToTfThreeVar(*result.Result.ID, zoneID, crn))
+	return resourceIBMCISDnsRecordUpdate(d, meta)
+
+}
+
+func resourceIBMCISDnsRecordRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		crn      string
+		zoneID   string
+		recordID string
+	)
+	sess, err := meta.(ClientSession).CisDNSRecordClientSession()
+	if err != nil {
+		return err
+	}
+
+	recordID, zoneID, crn, _ = convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return err
+	}
+	sess.Crn = core.StringPtr(crn)
+	sess.ZoneIdentifier = core.StringPtr(zoneID)
+
+	opt := sess.NewGetDnsRecordOptions(recordID)
+	result, response, err := sess.GetDnsRecord(opt)
+	if err != nil {
+		log.Printf("Error reading dns record: %s", response)
+		return err
+	}
+
+	d.Set(cisDNSRecordID, *result.Result.ID)
+	d.Set(cisZoneName, *result.Result.ZoneName)
+	d.Set(cisDNSRecordCreatedOn, *result.Result.CreatedOn)
+	d.Set(cisDNSRecordModifiedOn, *result.Result.ModifiedOn)
+	d.Set(cisDNSRecordName, *result.Result.Name)
+	d.Set(cisDNSRecordType, *result.Result.Type)
+	d.Set(cisDNSRecordContent, *result.Result.Content)
+	d.Set(cisDomainID, *result.Result.ZoneID)
```

Comment:
running terraform apply twice returns infrastructure change..
please set domainID as combination of <zoneID>:<cisID>

Reply:
ZoneID:cisID set appropriately as mentioned already.  

I tried to do ```terraform apply``` twice... it does not change anything. Please check the log below,

```bash
kumarganesan@Kumars-MBP$ tfa
ibm_cis_dns_record.test_dns_a_record: Creating...
ibm_cis_dns_record.test_dns_a_record: Creation complete after 5s [id=918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

a_record_output = {
  "cis_id" = "crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "content" = "1.2.3.4"
  "created_on" = "2020-08-24T06:42:15.404107Z"
  "domain_id" = "fd613592752217b573edbf741c893ecb"
  "id" = "918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "modified_on" = "2020-08-24T06:42:15.404107Z"
  "name" = "test-exmple.sdk.cistest-load.com"
  "proxiable" = true
  "proxied" = true
  "record_id" = "918df76aaea20008041073c33244bfa7"
  "ttl" = 1
  "type" = "A"
}

--------------------------------------------------------------------------------------------------------------------------------------------------
[Mon Aug 24 12:12:17 : ~/work/src/github.com/tf/examples/ibm-cis-example]
kumarganesan@Kumars-MBP$ tfa
ibm_cis_dns_record.test_dns_a_record: Refreshing state... [id=918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

a_record_output = {
  "cis_id" = "crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "content" = "1.2.3.4"
  "created_on" = "2020-08-24T06:42:15.404107Z"
  "domain_id" = "fd613592752217b573edbf741c893ecb"
  "id" = "918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "modified_on" = "2020-08-24T06:42:15.404107Z"
  "name" = "test-exmple.sdk.cistest-load.com"
  "proxiable" = true
  "proxied" = true
  "record_id" = "918df76aaea20008041073c33244bfa7"
  "ttl" = 1
  "type" = "A"
}
```